### PR TITLE
[SPARK-26985][CORE] Fix "access only some column of the all of columns " for big endian architecture

### PR DESCRIPTION
**Pull request by ketank-new**
## What changes were proposed in this pull request?

Changes are related to BIG ENDIAN system
This changes are done to 
1. identify s390x platform.
2. use byteorder to BIG_ENDIAN for big endian systems
3. changes for 2 are done in access functions putFloats() and putDouble()

## How was this patch tested?

Changes have been tested to build successfully on s390x as well x86 platform to make sure build is successful.

### CODE REVIEW

File: sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/OnHeapColumnVector.java [hunk 1]
```diff
@@ -396,7 +396,7 @@ public void putFloats(int rowId, int count, byte[] src, int srcIndex) {
       Platform.copyMemory(src, Platform.BYTE_ARRAY_OFFSET + srcIndex, floatData,
           Platform.DOUBLE_ARRAY_OFFSET + rowId * 4L, count * 4L);
     } else {
-      ByteBuffer bb = ByteBuffer.wrap(src).order(ByteOrder.LITTLE_ENDIAN);
+      ByteBuffer bb = ByteBuffer.wrap(src).order(ByteOrder.BIG_ENDIAN);
```
Comment (kiszk) [inline sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/OnHeapColumnVector.java]: Could you please let us know which test cases are failed due to this code on `s390x`?

**Comment by kiszk** [comment]
Could you please add JIRA number `[SPARK-....]` and `[core]` as a prefix of the title?

File: common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java [hunk 2]
```diff
@@ -51,6 +51,8 @@
       // Since java.nio.Bits.unaligned() doesn't return true on ppc (See JDK-8165231), but
       // ppc64 and ppc64le support it
       _unaligned = true;
+    } else if (arch.matches("s390x")) {
```
Comment (srowen) [inline common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java]: `.contains`? that isn't a regex. You can put it into the clause above and modify the comment

**Comment by srowen** [comment]
This needs a better title -- 'changes' isn't descriptive. What problem are you solving? It will need a JIRA if the change is more than simply including it as a big-endian platform, as the rest is pretty non-trivial

File: sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/OnHeapColumnVector.java (hunk 1 above)
Comment (gatorsmile) [reply sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/OnHeapColumnVector.java]: This change looks risky. Any reason?

Comment (kiszk) [reply sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/OnHeapColumnVector.java]: I agree. Once we know the test cases that cause failures, I will double-check by running them on another big endian environment.

**Comment by dongjoon-hyun** [comment]
ok to test

**Comment by dongjoon-hyun** [comment]
This seems to fail in Jenkins, doesn't it?

File: sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/OnHeapColumnVector.java (hunk 1 above)
Comment (ketank-new) [reply sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/OnHeapColumnVector.java]: below are the test suites which had some test failing initially and then passed on doing the above source code changes

spark/sql/core/src/test/scala/org/apache/spark/sql/execution/columnar/compression/PassThroughEncodingSuite.scala

spark/sql/core/src/test/scala/org/apache/spark/sql/execution/vectorized/ColumnVectorSuite.scala

spark/sql/core/src/test/scala/org/apache/spark/sql/execution/columnar/InMemoryColumnarQuerySuite.scala

spark/sql/core/src/test/scala/org/apache/spark/sql/DataFrameTungstenSuite.scala

**Comment by ketank-new** [comment]
below are the test suites which had some test failing initially and then passed on doing the above source code changes

spark/sql/core/src/test/scala/org/apache/spark/sql/execution/columnar/compression/PassThroughEncodingSuite.scala

spark/sql/core/src/test/scala/org/apache/spark/sql/execution/vectorized/ColumnVectorSuite.scala

spark/sql/core/src/test/scala/org/apache/spark/sql/execution/columnar/InMemoryColumnarQuerySuite.scala

spark/sql/core/src/test/scala/org/apache/spark/sql/DataFrameTungstenSuite.scala

File: sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/OnHeapColumnVector.java (hunk 1 above)
Comment (srowen) [reply sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/OnHeapColumnVector.java]: Hm, I guess the changes make sense? we don't test the big-endian branch on Jenkins, so I'm not surprised it still passes Jenkins. For a big-endian platform, indeed, wouldn't you have to read floats from the ByteBuffer as big-endian?

Comment (kiszk) [reply sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/OnHeapColumnVector.java]: Thanks, I will test them on ppc64 linux (BE).

**Comment by kiszk** [comment]
In a big-endian testing environment, if necessary, I could contact the usage of [this envionment](http://osuosl.org/services/powerdev/request_powerci/)

**Comment by HyukjinKwon** [comment]
@ketank-new, please file a JIRA with error message with problem analysis and describe how this PR fixes in PR description. Otherwise, no one knows what's going on about what you faced.

**Comment by ketank-new** [comment]
@HyukjinKwon : I do not mind raising a new JIRA for the above changes
But let me inform you that we have been continuously in discussion on JIRA regarding this earlier too
Please refer SPARK-26985

**Comment by HyukjinKwon** [comment]
If that's the exact same problem, please link that JIRA to the PR title (see https://spark.apache.org/contributing.html).

Also, please clarify why changing from little endian to big endian is safe in little endian OSes.

**Comment by ketank-new** [comment]
> Could you please add JIRA number `[SPARK-....]` and `[core]` as a prefix of the title?

done

**Comment by ketank-new** [comment]
> If that's the exact same problem, please link that JIRA to the PR title (see https://spark.apache.org/contributing.html).

Linked by changing the Title above.. were in the JIRA number is seen
> 
> Also, please clarify why changing from little endian to big endian is safe in little endian OSes.

Clarification for the changes
putFloats() and putDoubles() from files OffHeapColumnVector.java and OnHeapColumnVector.java
do get called whenever test cases use float and double data respectively.

If you check the definitions of putFloat() and putDouble() for a BIG_ENDIAN system, the control moves into the else block,within the else block the byteorder which is set is LITTLE_ENDIAN which is exactly opposite to what is expected on a BIG_ENDIAN system

changing this byteorder to BIG_ENDIAN represents float and double values as expected on a BIG_ENDIAN processor and hence further helps in passing the written test cases.

With this changes done and running the test cases on LITTLE_ENDIAN moves the control to if block and thereby also passes test cases for LITTLE_ENDIAN system.

Thereby i conclude that the changes are working and test for both types of systems that is LITTLE_ENDIAN and BIG_ENDIAN

**Comment by viirya** [comment]
@ketank-new Can you update the PR description like above clarification of changes? It can explain what this PR wants to fix.

**Comment by ketank-new** [comment]
> @ketank-new Can you update the PR description like above clarification of changes? It can explain what this PR wants to fix.

done

File: common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java (hunk 2 above)
Comment (srowen) [reply common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java]: This can still be changed

**Review by srowen — CHANGES_REQUESTED** [review]
I'll fix the title for you; PRs need descriptive title

**Comment by kiszk** [comment]
I double-checked some of tests. I confirmed this PR fixes existing failures on big-endian platform. Thank you very much.

**Before**
```
$ uname -a
Linux ppc64be.novalocal 4.5.7-300.fc24.ppc64 #1 SMP Fri Jun 10 20:29:32 UTC 2016 ppc64 ppc64 ppc64 GNU/Linux
$ build/sbt "sql/testOnly  *.ColumnVectorSuite"
...
[info] ColumnVectorSuite:
[info] - boolean (194 milliseconds)
[info] - byte (4 milliseconds)
[info] - short (4 milliseconds)
[info] - int (4 milliseconds)
[info] - date (4 milliseconds)
[info] - long (4 milliseconds)
[info] - timestamp (5 milliseconds)
[info] - float (4 milliseconds)
[info] - double (4 milliseconds)
[info] - string (14 milliseconds)
[info] - binary (8 milliseconds)
[info] - mutable ColumnarRow (4 milliseconds)
[info] - array (1 millisecond)
[info] - struct (1 millisecond)
[info] - [SPARK-22092] off-heap column vector reallocation corrupts array data (3 milliseconds)
[info] - [SPARK-22092] off-heap column vector reallocation corrupts struct nullability (3 milliseconds)
[info] - CachedBatch boolean Apis (46 milliseconds)
[info] - CachedBatch byte Apis (7 milliseconds)
[info] - CachedBatch short Apis (8 milliseconds)
[info] - CachedBatch int Apis (9 milliseconds)
[info] - CachedBatch long Apis (8 milliseconds)
[info] - CachedBatch float Apis *** FAILED *** (30 milliseconds)
[info]   4.6006E-41 did not equal 1.0 (ColumnVectorSuite.scala:424)
[info]   org.scalatest.exceptions.TestFailedException:
[info]   at org.scalatest.Assertions.newAssertionFailedException(Assertions.scala:528)
[info]   at org.scalatest.Assertions.newAssertionFailedException$(Assertions.scala:527)
[info]   at org.scalatest.FunSuite.newAssertionFailedException(FunSuite.scala:1560)
[info]   at org.scalatest.Assertions$AssertionsHelper.macroAssert(Assertions.scala:501)
[info]   at org.apache.spark.sql.execution.vectorized.ColumnVectorSuite.$anonfun$new$71(ColumnVectorSuite.scala:424)
[info]   at org.apache.spark.sql.execution.vectorized.ColumnVectorSuite.$anonfun$new$71$adapted(ColumnVectorSuite.scala:422)
[info]   at scala.collection.immutable.Range.foreach(Range.scala:158)
[info]   at org.apache.spark.sql.execution.vectorized.ColumnVectorSuite.$anonfun$new$70(ColumnVectorSuite.scala:422)
[info]   at org.apache.spark.sql.execution.vectorized.ColumnVectorSuite.$anonfun$new$70$adapted(ColumnVectorSuite.scala:417)
[info]   at org.apache.spark.sql.execution.vectorized.ColumnVectorSuite.withVector(ColumnVectorSuite.scala:34)
[info]   at org.apache.spark.sql.execution.vectorized.ColumnVectorSuite.withVectors(ColumnVectorSuite.scala:41)
[info]   at org.apache.spark.sql.execution.vectorized.ColumnVectorSuite.$anonfun$new$68(ColumnVectorSuite.scala:417)
[info]   at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.FunSuiteLike$$anon$1.apply(FunSuiteLike.scala:186)
[info]   at org.apache.spark.SparkFunSuite.withFixture(SparkFunSuite.scala:149)
[info]   at org.scalatest.FunSuiteLike.invokeWithFixture$1(FunSuiteLike.scala:184)
[info]   at org.scalatest.FunSuiteLike.$anonfun$runTest$1(FunSuiteLike.scala:196)
[info]   at org.scalatest.SuperEngine.runTestImpl(Engine.scala:289)
[info]   at org.scalatest.FunSuiteLike.runTest(FunSuiteLike.scala:196)
[info]   at org.scalatest.FunSuiteLike.runTest$(FunSuiteLike.scala:178)
[info]   at org.apache.spark.SparkFunSuite.org$scalatest$BeforeAndAfterEach$$super$runTest(SparkFunSuite.scala:56)
[info]   at org.scalatest.BeforeAndAfterEach.runTest(BeforeAndAfterEach.scala:221)
[info]   at org.scalatest.BeforeAndAfterEach.runTest$(BeforeAndAfterEach.scala:214)
[info]   at org.apache.spark.SparkFunSuite.runTest(SparkFunSuite.scala:56)
[info]   at org.scalatest.FunSuiteLike.$anonfun$runTests$1(FunSuiteLike.scala:229)
[info]   at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:396)
[info]   at scala.collection.immutable.List.foreach(List.scala:392)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:384)
[info]   at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:379)
[info]   at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:461)
[info]   at org.scalatest.FunSuiteLike.runTests(FunSuiteLike.scala:229)
[info]   at org.scalatest.FunSuiteLike.runTests$(FunSuiteLike.scala:228)
[info]   at org.scalatest.FunSuite.runTests(FunSuite.scala:1560)
[info]   at org.scalatest.Suite.run(Suite.scala:1147)
[info]   at org.scalatest.Suite.run$(Suite.scala:1129)
[info]   at org.scalatest.FunSuite.org$scalatest$FunSuiteLike$$super$run(FunSuite.scala:1560)
[info]   at org.scalatest.FunSuiteLike.$anonfun$run$1(FunSuiteLike.scala:233)
[info]   at org.scalatest.SuperEngine.runImpl(Engine.scala:521)
[info]   at org.scalatest.FunSuiteLike.run(FunSuiteLike.scala:233)
[info]   at org.scalatest.FunSuiteLike.run$(FunSuiteLike.scala:232)
[info]   at org.apache.spark.SparkFunSuite.org$scalatest$BeforeAndAfterAll$$super$run(SparkFunSuite.scala:56)
[info]   at org.scalatest.BeforeAndAfterAll.liftedTree1$1(BeforeAndAfterAll.scala:213)
[info]   at org.scalatest.BeforeAndAfterAll.run(BeforeAndAfterAll.scala:210)
[info]   at org.scalatest.BeforeAndAfterAll.run$(BeforeAndAfterAll.scala:208)
[info]   at org.apache.spark.SparkFunSuite.run(SparkFunSuite.scala:56)
[info]   at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:314)
[info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:507)
[info]   at sbt.ForkMain$Run$2.call(ForkMain.java:296)
[info]   at sbt.ForkMain$Run$2.call(ForkMain.java:286)
[info]   at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[info]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[info]   at java.lang.Thread.run(Thread.java:748)
[info] - CachedBatch double Apis *** FAILED *** (11 milliseconds)
[info]   3.03865E-319 did not equal 1.0 (ColumnVectorSuite.scala:448)
[info]   org.scalatest.exceptions.TestFailedException:
[info]   at org.scalatest.Assertions.newAssertionFailedException(Assertions.scala:528)
[info]   at org.scalatest.Assertions.newAssertionFailedException$(Assertions.scala:527)
[info]   at org.scalatest.FunSuite.newAssertionFailedException(FunSuite.scala:1560)
[info]   at org.scalatest.Assertions$AssertionsHelper.macroAssert(Assertions.scala:501)
[info]   at org.apache.spark.sql.execution.vectorized.ColumnVectorSuite.$anonfun$new$75(ColumnVectorSuite.scala:448)
[info]   at org.apache.spark.sql.execution.vectorized.ColumnVectorSuite.$anonfun$new$75$adapted(ColumnVectorSuite.scala:446)
[info]   at scala.collection.immutable.Range.foreach(Range.scala:158)
[info]   at org.apache.spark.sql.execution.vectorized.ColumnVectorSuite.$anonfun$new$74(ColumnVectorSuite.scala:446)
[info]   at org.apache.spark.sql.execution.vectorized.ColumnVectorSuite.$anonfun$new$74$adapted(ColumnVectorSuite.scala:441)
[info]   at org.apache.spark.sql.execution.vectorized.ColumnVectorSuite.withVector(ColumnVectorSuite.scala:34)
[info]   at org.apache.spark.sql.execution.vectorized.ColumnVectorSuite.withVectors(ColumnVectorSuite.scala:41)
[info]   at org.apache.spark.sql.execution.vectorized.ColumnVectorSuite.$anonfun$new$72(ColumnVectorSuite.scala:441)
[info]   at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.FunSuiteLike$$anon$1.apply(FunSuiteLike.scala:186)
[info]   at org.apache.spark.SparkFunSuite.withFixture(SparkFunSuite.scala:149)
[info]   at org.scalatest.FunSuiteLike.invokeWithFixture$1(FunSuiteLike.scala:184)
[info]   at org.scalatest.FunSuiteLike.$anonfun$runTest$1(FunSuiteLike.scala:196)
[info]   at org.scalatest.SuperEngine.runTestImpl(Engine.scala:289)
[info]   at org.scalatest.FunSuiteLike.runTest(FunSuiteLike.scala:196)
[info]   at org.scalatest.FunSuiteLike.runTest$(FunSuiteLike.scala:178)
[info]   at org.apache.spark.SparkFunSuite.org$scalatest$BeforeAndAfterEach$$super$runTest(SparkFunSuite.scala:56)
[info]   at org.scalatest.BeforeAndAfterEach.runTest(BeforeAndAfterEach.scala:221)
[info]   at org.scalatest.BeforeAndAfterEach.runTest$(BeforeAndAfterEach.scala:214)
[info]   at org.apache.spark.SparkFunSuite.runTest(SparkFunSuite.scala:56)
[info]   at org.scalatest.FunSuiteLike.$anonfun$runTests$1(FunSuiteLike.scala:229)
[info]   at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:396)
[info]   at scala.collection.immutable.List.foreach(List.scala:392)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:384)
[info]   at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:379)
[info]   at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:461)
[info]   at org.scalatest.FunSuiteLike.runTests(FunSuiteLike.scala:229)
[info]   at org.scalatest.FunSuiteLike.runTests$(FunSuiteLike.scala:228)
[info]   at org.scalatest.FunSuite.runTests(FunSuite.scala:1560)
[info]   at org.scalatest.Suite.run(Suite.scala:1147)
[info]   at org.scalatest.Suite.run$(Suite.scala:1129)
[info]   at org.scalatest.FunSuite.org$scalatest$FunSuiteLike$$super$run(FunSuite.scala:1560)
[info]   at org.scalatest.FunSuiteLike.$anonfun$run$1(FunSuiteLike.scala:233)
[info]   at org.scalatest.SuperEngine.runImpl(Engine.scala:521)
[info]   at org.scalatest.FunSuiteLike.run(FunSuiteLike.scala:233)
[info]   at org.scalatest.FunSuiteLike.run$(FunSuiteLike.scala:232)
[info]   at org.apache.spark.SparkFunSuite.org$scalatest$BeforeAndAfterAll$$super$run(SparkFunSuite.scala:56)
[info]   at org.scalatest.BeforeAndAfterAll.liftedTree1$1(BeforeAndAfterAll.scala:213)
[info]   at org.scalatest.BeforeAndAfterAll.run(BeforeAndAfterAll.scala:210)
[info]   at org.scalatest.BeforeAndAfterAll.run$(BeforeAndAfterAll.scala:208)
[info]   at org.apache.spark.SparkFunSuite.run(SparkFunSuite.scala:56)
[info]   at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:314)
[info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:507)
[info]   at sbt.ForkMain$Run$2.call(ForkMain.java:296)
[info]   at sbt.ForkMain$Run$2.call(ForkMain.java:286)
[info]   at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[info]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[info]   at java.lang.Thread.run(Thread.java:748)
[info] ScalaTest
[info] Run completed in 6 seconds, 911 milliseconds.
[info] Total number of tests run: 23
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 21, failed 2, canceled 0, ignored 0, pending 0
[info] *** 2 TESTS FAILED ***
[error] Failed: Total 23, Failed 2, Errors 0, Passed 21
[error] Failed tests:
[error] 	org.apache.spark.sql.execution.vectorized.ColumnVectorSuite
[error] (sql/test:testOnly) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 700 s, completed Jun 9, 2019 11:23:46 PM
```

**After**
```
$ uname -a
Linux ppc64be.novalocal 4.5.7-300.fc24.ppc64 #1 SMP Fri Jun 10 20:29:32 UTC 2016 ppc64 ppc64 ppc64 GNU/Linux
$ build/sbt "sql/testOnly  *.ColumnVectorSuite"
...
[info] ColumnVectorSuite:
[info] - boolean (187 milliseconds)
[info] - byte (6 milliseconds)
[info] - short (6 milliseconds)
[info] - int (5 milliseconds)
[info] - date (4 milliseconds)
[info] - long (4 milliseconds)
[info] - timestamp (4 milliseconds)
[info] - float (4 milliseconds)
[info] - double (4 milliseconds)
[info] - string (13 milliseconds)
[info] - binary (8 milliseconds)
[info] - mutable ColumnarRow (5 milliseconds)
[info] - array (1 millisecond)
[info] - struct (2 milliseconds)
[info] - [SPARK-22092] off-heap column vector reallocation corrupts array data (4 milliseconds)
[info] - [SPARK-22092] off-heap column vector reallocation corrupts struct nullability (3 milliseconds)
[info] - CachedBatch boolean Apis (45 milliseconds)
[info] - CachedBatch byte Apis (7 milliseconds)
[info] - CachedBatch short Apis (7 milliseconds)
[info] - CachedBatch int Apis (9 milliseconds)
[info] - CachedBatch long Apis (8 milliseconds)
[info] - CachedBatch float Apis (8 milliseconds)
[info] - CachedBatch double Apis (7 milliseconds)
[info] ScalaTest
[info] Run completed in 6 seconds, 733 milliseconds.
[info] Total number of tests run: 23
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 23, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[info] Passed: Total 23, Failed 0, Errors 0, Passed 23
[success] Total time: 38 s, completed Jun 9, 2019 11:52:54 PM
```

File: common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java (hunk 2 above)
Comment (ketank-new) [reply common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java]: @srowen  : did not get your <.contains> comment what exactly you are trying to say
I have intentionally written a else if blk to keep processor specific changes in seperate block.

are you asking me to add "s390x" in the same "ppc64le" blk ... let me know

Comment (srowen) [reply common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java]: Yes, keep them together. And it does not need to be a regex with `matches`; just use `contains`.

**Comment by dongjoon-hyun** [comment]
Thank you for confirming, @kiszk .

**Comment by dongjoon-hyun** [comment]
@ketank-new Could you address the comments and resolve the conflicts?

File: common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java (hunk 2 above)
Comment (ketank-new) [reply common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java]: > @ketank-new Could you address the comments and resolve the conflicts?

sure

**Comment by ketank-new** [comment]
> @ketank-new Could you address the comments and resolve the conflicts?

done

**Comment by kiszk** [comment]
retest this please

File: common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java [hunk 3]
```diff
@@ -44,6 +44,28 @@
   public static final int DOUBLE_ARRAY_OFFSET;
 
   private static final boolean unaligned;
+  static {
```
Comment (viirya) [inline common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java]: Ur, #24825 touched `unaligned` initialization and just got merged. Please rebase with the master.

Comment (ketank-new) [reply common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java]: @viirya  : are u asked me to rebase with master for Platform.java changes??

Comment (viirya) [reply common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java]: Yes, otherwise it can't be built.

File: docs/sql-migration-guide-upgrade.md [hunk 4]
```diff
@@ -23,6 +23,7 @@ license: |
 {:toc}
 
 ## Upgrading From Spark SQL 2.4 to 3.0
+  - In Spark version 2.4 and earlier, SQL queries such as `FROM <table>` or `FROM <table> UNION ALL FROM <table>` are supported by accident. In hive-style `FROM <table> SELECT <expr>`, the `SELECT` clause is not negligible. Neither Hive nor Presto support this syntax. Therefore we will treat these queries as invalid since Spark 3.0.
```
Comment (srowen) [inline docs/sql-migration-guide-upgrade.md]: I think you included a bunch of unrelated changes. Can you revert or open a new PR?

**Review by srowen — CHANGES_REQUESTED** [review]

File: docs/sql-migration-guide-upgrade.md (hunk 4 above)
Comment (ketank-new) [reply docs/sql-migration-guide-upgrade.md]: yes .. looks like .. i better open a new PR

**Comment by HyukjinKwon** [comment]
Closing per #24861

**Comment by HyukjinKwon** [comment]
Closing per #24861

**Comment by HyukjinKwon** [comment]
My cat reopened .. closing again.
